enterprise/internal/insights/compression/commits.go

```diff
@@ -0,0 +1,157 @@
+package compression
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"time"
+
+	"github.com/keegancsmith/sqlf"
+
+	"github.com/sourcegraph/sourcegraph/internal/api"
+	"github.com/sourcegraph/sourcegraph/internal/database/basestore"
+	"github.com/sourcegraph/sourcegraph/internal/database/dbutil"
+	"github.com/sourcegraph/sourcegraph/internal/vcs/git"
+)
+
+type DBCommitStore struct {
+	*basestore.Store
+}
+
+type CommitStore interface {
+	Save(ctx context.Context, id api.RepoID, commit *git.Commit) error
+	Get(ctx context.Context, id api.RepoID, start time.Time, end time.Time) ([]CommitStamp, error)
+	GetMetadata(ctx context.Context, id api.RepoID) (CommitIndexMetadata, error)
+	UpsertMetadataStamp(ctx context.Context, id api.RepoID) (CommitIndexMetadata, error)
+	InsertCommits(ctx context.Context, id api.RepoID, commits []*git.Commit) error
+}
+
+func NewCommitStore(db dbutil.DB) *DBCommitStore {
+	return &DBCommitStore{
+		Store: basestore.NewWithDB(db, sql.TxOptions{}),
+	}
+}
+
+func (c *DBCommitStore) With(other basestore.ShareableStore) *DBCommitStore {
+	return &DBCommitStore{Store: c.Store.With(other)}
+}
+
+func (c *DBCommitStore) Transact(ctx context.Context) (*DBCommitStore, error) {
+	txBase, err := c.Store.Transact(ctx)
+	return &DBCommitStore{Store: txBase}, err
+}
+
+func (c *DBCommitStore) Save(ctx context.Context, id api.RepoID, commit *git.Commit) error {
+	ref := commit.ID
+	err := c.Exec(ctx, sqlf.Sprintf(insertCommitIndexStr, id, ref, commit.Committer.Date))
+	if err != nil {
+		return fmt.Errorf("error saving commit for repo_id: %v ref %v: %w", id, ref, err)
+	}
+
+	return nil
+}
+
+func (c *DBCommitStore) InsertCommits(ctx context.Context, id api.RepoID, commits []*git.Commit) error {
+	tx, err := c.Transact(ctx)
+	if err != nil {
+		return err
+	}
+
+	defer tx.Store.Done(err)
+
+	for _, commit := range commits {
+
+		err = tx.Save(ctx, id, commit)
+		if err != nil {
+			return err
+		}
```

Comment:
```suggestion
		if err := tx.Save(ctx, id, commit); err != nil {
			return err
		}
```

Nit: I tend to prefer limiting the scope of `err` where possible, especially if the only place it's used is in a subsequent conditional return.

No big issue here, just wanted to give a preference (our codebase has a scattering of a bunch of different styles here).